packages/langchain/src/orchestration/util.ts

```diff
@@ -174,3 +194,81 @@ export function mapOutputToChatResult(
     }
   };
 }
+
+/**
+ * Converts orchestration stream chunk to a LangChain message chunk.
+ * @param chunk- The orchestration stream chunk.
+ * @returns An {@link AIMessageChunk}
+ * @internal
+ */
+export function mapOrchestrationChunkToLangChainMessageChunk(
+  chunk: OrchestrationStreamChunkResponse
+): AIMessageChunk {
+  const { module_results, request_id } = chunk.data;
+  const content = chunk.getDeltaContent() ?? '';
+  const toolCallChunks = chunk.getDeltaToolCallChunks();
+
+  const additional_kwargs: Record<string, unknown> = {
+    module_results,
+    request_id
+  };
+
+  let tool_call_chunks: ToolCallChunk[] = [];
+  if (Array.isArray(toolCallChunks)) {
+    tool_call_chunks = mapOrchestrationToLangchainToolCallChunk(toolCallChunks);
+  }
+  // Use AIMessageChunk to represent message chunks for roles like 'tool' and 'user' too
+  return new AIMessageChunk({ content, additional_kwargs, tool_call_chunks });
+}
+
+/**
+ * Sets finish reason on a LangChain message chunk if available.
+ * @param messageChunk - The LangChain message chunk to update.
+ * @param finishReason - The finish reason from the response.
+ * @internal
+ */
+export function setFinishReason(
+  messageChunk: AIMessageChunk,
+  finishReason: string | undefined
+): void {
+  if (finishReason) {
+    messageChunk.response_metadata.finish_reason = finishReason;
+  }
+}
+
+/**
+ * Sets usage metadata on a message chunk if available.
+ * @param messageChunk - The LangChain message chunk to update.
+ * @param tokenUsage - The token usage information.
+ * @internal
+ */
+export function setTokenUsage(
+  messageChunk: AIMessageChunk,
+  tokenUsage: TokenUsage | undefined
+): void {
+  if (tokenUsage) {
+    messageChunk.usage_metadata = {
+      input_tokens: tokenUsage.prompt_tokens,
+      output_tokens: tokenUsage.completion_tokens,
+      total_tokens: tokenUsage.total_tokens
+    };
+    messageChunk.response_metadata.token_usage = tokenUsage;
+  }
+}
+
+/**
+ * Computes token indices for a chunk of the stream response.
+ * @param chunk - A chunk of the stream response.
+ * @returns An object with prompt and completion indices.
+ * @internal
+ */
+export function computeTokenIndices(chunk: OrchestrationStreamChunkResponse): {
+  prompt: number;
+  completion: number;
+} {
+  return {
+    // Indicates the token is part of the first prompt
+    prompt: 0,
+    completion: chunk.data.orchestration_result?.choices[0]?.index ?? 0
```

Comment:
[req] Here do not assume we only have one choice. We do not know if the first element in the chunk is for choice 0.

Reply:
Here if we only have one choice, why don't we just set completion to 0?